presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1666,6 +2273,173 @@ public void dropView(ConnectorSession session, SchemaTableName viewName)
+    @Override
+    public void createMaterializedView(ConnectorSession session, ConnectorTableMetadata viewMetadata, ConnectorMaterializedViewDefinition viewDefinition, boolean ignoreExisting)
+    {
+        if (isExternalTable(viewMetadata.getProperties())) {
+            throw new PrestoException(INVALID_TABLE_PROPERTY, "Specifying external location for materialized view is not supported.");
+        }
+
+        Table basicTable = prepareTable(session, viewMetadata, MATERIALIZED_VIEW);
+        viewDefinition = new ConnectorMaterializedViewDefinition(
+                viewDefinition.getOriginalSql(),
+                viewDefinition.getSchema(),
+                viewDefinition.getTable(),
+                viewDefinition.getBaseTables(),
+                viewDefinition.getOwner(),
+                viewDefinition.getColumnMappings(),
+                Optional.of(getPartitionedBy(viewMetadata.getProperties())));
+        Map<String, String> parameters = ImmutableMap.<String, String>builder()
+                .putAll(basicTable.getParameters())
+                .put(PRESTO_MATERIALIZED_VIEW_FLAG, "true")
+                .build();
+        Table viewTable = Table.builder(basicTable)
+                .setParameters(parameters)
+                .setViewOriginalText(Optional.of(encodeMaterializedViewData(MATERIALIZED_VIEW_JSON_CODEC.toJson(viewDefinition))))
+                .setViewExpandedText(Optional.of("/* Presto Materialized View */"))
+                .build();
+
+        MetastoreContext metastoreContext = new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource(), getMetastoreHeaders(session));
+        validateMaterializedViewPartitionColumns(metastore, metastoreContext, viewTable, viewDefinition);
+
+        try {
+            PrincipalPrivileges principalPrivileges = buildInitialPrivilegeSet(viewTable.getOwner());
+            metastore.createTable(
+                    session,
+                    viewTable,
+                    principalPrivileges,
+                    Optional.empty(),
+                    ignoreExisting,
+                    new PartitionStatistics(createEmptyStatistics(), ImmutableMap.of()));
+        }
+        catch (TableAlreadyExistsException e) {
+            throw new MaterializedViewAlreadyExistsException(e.getTableName());
+        }
+    }
+
+    @Override
+    public void dropMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        Optional<ConnectorMaterializedViewDefinition> view = getMaterializedView(session, viewName);
+        if (!view.isPresent()) {
+            throw new MaterializedViewNotFoundException(viewName);
+        }
+
+        try {
+            metastore.dropTable(
+                    new HdfsContext(session, viewName.getSchemaName(), viewName.getTableName()),
+                    viewName.getSchemaName(),
+                    viewName.getTableName());
+        }
+        catch (TableNotFoundException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)